packages/suite-base/src/providers/CurrentLayoutProvider/index.tsx

```diff
@@ -259,12 +266,18 @@ export default function CurrentLayoutProvider({ children }: React.PropsWithChild
       return;
     }
 
+    // Try to load default layouts, before checking to add the fallback "Default".
+    await loadDefaultLayouts(layoutManager, loaders);
+
     // Retreive the selected layout id from the user's profile. If there's no layout specified
     // or we can't load it then save and select a default layout.
     const { currentLayoutId } = await getUserProfile();
     const layout = currentLayoutId ? await layoutManager.getLayout(currentLayoutId) : undefined;
+    const layouts = await layoutManager.getLayouts();
```

Comment:
I'd say you may be consulting layouts unnecessarily when currentLayoutId exists.
So, a suggestiong is split the logic below.

If layout -> setSelectedLayoutId and return void.
Then, continue with the remaining cases...
getLayouts(), etc.